### ISSUE TITLE
Cannot add elements to a Collection declared as Variant

### ISSUE THREAD

**Issue by cristianbuse** [WaynePhillipsEA/twinbasic#158]
**Describe the bug**
Cannot add elements to a Collection declared as Variant. Runtime error

**To Reproduce**
```VBA
Module Test
    Public Sub Main()
        Dim coll As Variant
        
        Set coll = New Collection
        coll.Add 1                              'tB runtime error here
    End Sub
End Module
```

**Expected behavior**
No runtime error should occur. Same code works in VBA.

**Screenshots**
![test](https://i.stack.imgur.com/EgXf7.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 Enterprise V1909
 - twinBASIC compiler v0.9.2720

**Additional context**
It seems like this issue doesn't occur on a tB class. The following code works just fine:  
```VBA
Module Test
    Public Sub Main()
        Dim coll As Variant
        
        Set coll = New Class1
        coll.Add 1
    End Sub
End Module

Class Class1
    Sub Add(v)
    End Sub
End Class
```

This might be related to issue #157

**Comment by WaynePhillipsEA** [comment]
@cristianbuse Just a little note to say thanks for all of the bug reports!  They're really going to help get the edge cases sorted.  Once the next update is out in a few days, I'll be on to them!

**Comment by WaynePhillipsEA** [comment]
The VBA.Collection and VBA.ErrObject implementations are currently running off an early prototype implementation for IDispatch, causing this issue and #153.  In a few days I'll rework them to work off the stable IDispatch implementations that we now use for normal tB classes.

**Comment by WaynePhillipsEA** [comment]
Although we haven't yet unified the implementation of IDispatch, we have fixed this bug in the old implementation for now.  Fixed in v0.10.833, thanks for reporting.
